qa-tests-backend/src/test/groovy/IntegrationsSplunkViolationsTest.groovy

```diff
@@ -134,19 +134,60 @@ class IntegrationsSplunkViolationsTest extends BaseSpecification {
             }
             hasNetworkViolation = results.any { isNetworkViolation(it) }
             hasProcessViolation = results.any { isProcessViolation(it) }
+            log.debug "Found violations in Splunk: \n${results}"
+            log.debug "hasNetworkViolation: ${hasNetworkViolation}\nhasProcessViolation: ${hasProcessViolation}"
             if (hasNetworkViolation && hasProcessViolation) {
-                log.info "Success!"
+                log.info "Success finding Network and Process Violations in Splunk!"
+                break
+            }
+        }
+
+        log.info "Starting conversion of ACS violations to Splunk alerts"
+        postToSplunk(port, "/services/saved/searches/" + SPLUNK_TA_CONVERSION_JOB_NAME +"/dispatch", [
+                "dispatch.now": "true",
+                "force_dispatch": "true",
+        ])
```

Comment:
If I'm not mistaken this will trigger the conversion, right? Does that mean that the logic inside the splunk TA that will trigger this conversion automatically is not tested?

I wonder if just setting the interval to `1m` (for example) and waiting in the next block would be sufficient without having to manually trigger the conversion.

Reply:
Correct, this triggers the conversion. 
I originally set this to 1m [in this commit](https://github.com/stackrox/stackrox/pull/11848/commits/405c477de9a304af25478ae2f87b81593ad0c7e2#diff-b5d8e8c78f856dbec06efbf9fdf975540853d6691648c5071a853559e9e47a5bL103), but after discussions in last weeks demo meeting, I reverted that change. 
It was argued that changing the cron would bring this further away from prod use.

I am triggering the same conversion job that is the cronjob. I would argue that we shouldn't do functional testing for the Splunk cronjob. We could introduce a third loop in the middle that just waits for the cronjob to have run, although I don't know whether we really should blow up the test runtime any further

Reply:
I don't have enough knowledge on the Splunk TA but they way I see it is:

- If the cronjob scheduler is something that splunk offers (not our code) and simply calls our conversion function, then I think it's ok to trigger the conversion manually.
- If the cronjob is our code, then I think it would be nice to verify that it's working by waiting until it triggers the conversion.

Reply:
It is the first one. Splunk offers its own cron scheduler that calls a custom conversion function that we have built :smile: